packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Dispatch } from 'redux'
+import { useHistory, useLocation } from 'react-router'
+import {
+  Loader,
+  Pagination,
+  Table,
+  Button,
+  FlexContainerResponsive,
+  Helper,
+  infoText,
+  FlexContainerBasic,
+} from '@reapit/elements'
+import { revisionDetailRequestData } from '@/actions/revision-detail'
+import routes from '@/constants/routes'
+import { REVISIONS_PER_PAGE } from '@/constants/paginator'
+import { appDetailRequestData } from '@/actions/app-detail'
+import { ApprovalModel, AppRevisionModel } from '@reapit/foundations-ts-definitions'
+import { AppDetailModel } from '@/types/marketplace-api-schema'
+import { selectApp } from '@/selector/app-detail'
+import { selectAppRevision } from '@/selector/app-revisions'
+import AdminApprovalModal from '@/components/ui/admin-approval-modal'
+import { selectAdminApprovalsState, selectWaitingApprovalData } from '@/selector/admin'
+
+export const handleCloseModal = ({ setIsModalOpen }) => () => setIsModalOpen(false)
+
+export const handleOnPageChange = history => page => history.push(`${routes.ADMIN_APPROVALS}?page=${page}`)
+
+export const Content = ({
+  loading,
+  waitingApprovalList,
+  tableColumns,
+}: {
+  loading: Boolean
+  waitingApprovalList: ApprovalModel[]
+  tableColumns: any
+}) => {
+  if (loading) {
+    return (
+      <div className="pin absolute flex items-center justify-center">
+        <Loader />
+      </div>
+    )
+  }
+
+  if (!loading && !waitingApprovalList.length) {
+    return (
+      <FlexContainerBasic hasPadding flexColumn>
+        <Helper variant="info">{infoText('ADMIN_APPROVALS_EMPTY')}</Helper>
+      </FlexContainerBasic>
+    )
+  }
+
+  return (
+    <div>
+      <Table scrollable={true} loading={false} data={waitingApprovalList} columns={tableColumns} />
+    </div>
+  )
+}
+
+export type RenderId = {
+  page: number
+}
+
+export const renderId = ({ page }: RenderId) => ({ row: { index } }) => {
+  const pageNoTimesRevsions = (page - 1) * REVISIONS_PER_PAGE
+  return <div>{pageNoTimesRevsions + index + 1}</div>
+}
+
+export type HandleViewDetailOnClick = {
+  dispatch: Dispatch<any>
+  appId: string
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+  appRevisionId: string
+  currentRevisionId?: string
+  currentAppId?: string
+}
+
+export const handleViewDetailOnClick = ({
```

Comment:
Please add params type definition

Reply:
This one also have type definition for params